### PR TITLE
Lps 72500

### PR DESCRIPTION
**Pull request by shuyangzhou**

### CODE REVIEW

**Comment by rotty3000** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: bundles/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/HttpSessionTrackerUtil.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/**
```
Comment (rotty3000) [inline bundles/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/HttpSessionTrackerUtil.java]: this class does not appear to be used anywhere! Can we remove it?

**Comment by shuyangzhou** [comment]
It is used in portal.

File: bundles/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/HttpSessionTrackerUtil.java (hunk 1 above)
Comment (shuyangzhou) [reply bundles/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/HttpSessionTrackerUtil.java]: It is here https://github.com/liferay/liferay-portal/blob/master/modules/apps/static/portal-osgi-web/portal-osgi-web-wab-extender/src/main/java/com/liferay/portal/osgi/web/wab/extender/internal/adapter/HttpAdapter.java#L169

**Comment by rotty3000** [comment]
We'll need to implement this differently. We can't just expose this static method like this.

**Comment by shuyangzhou** [comment]
It was done this way, because HttpSessionTracker is internal. If you want to get rid of that util, we will have to expose HttpSessionTracker, however only the clear method is needed, exposing it as a whole won't really make sense.

**Comment by rotty3000** [comment]
I think what's needed here is an SPI mechanism that Liferay can implement to hook into this functionality. A static method is not usable in several other deployment scenarios.

**Comment by shuyangzhou** [comment]
Ok, that should be fine, we just need a way to clear it. It does not really matter in which way. Can you make that change and apply it to the wab-extender? That will save one round of reviews.

**Comment by rotty3000** [comment]
it would have been nice if tests was written for these changes.

**Comment by rotty3000** [comment]
See https://bugs.eclipse.org/bugs/show_bug.cgi?id=530063
We'll have to wait and see when it gets in.

**Pull request closed without merging**